synthesizer/process/src/stack/deploy.rs

```diff
@@ -67,6 +68,11 @@ impl<N: Network> Stack<N> {
 
         // Ensure the deployment is ordered.
         deployment.check_is_ordered()?;
+        // Ensure that edition in the stack and deployment matches.
```

Comment:
While this check is not incorrect, it's not practical since a user may want to verify deployments without needing to use the exact edition.